src/engine.py

```
@@ -17562,27 +17775,51 @@ async def _reload_handler(self, request):
                     "message": "Failed to reload configuration",
                     "errors": errors
                 }, status=500)
+
+            # Build every new-call runtime object off-side. Nothing running is
+            # mutated unless both tool construction and orchestration validation
+            # succeed completely.
+            try:
+                new_tool_generation = self._build_tool_generation(new_config)
+                cfg_dict = new_config.dict() if hasattr(new_config, "dict") else new_config.__dict__
+                new_transport_orchestrator = TransportOrchestrator(cfg_dict)
+            except Exception as e:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)